### PR TITLE
Use exceptions

### PR DESCRIPTION
**Pull request by benrudolph**
It's pretty painful working with Jython 2.5.3. It means I can't use a lot of nice python things like `with` and modules have been a huge thorn in my side which is why there is no `tests` directory. The Jython classpath gets really messed. Jython 2.7 just got released and I think we should try to upgrade. I was able to more easily create modules and I don't have to think about the differences between python 2.5 and 2.7

@esoergel

### CODE REVIEW

**Comment by czue** [comment]
nice cleanup!

File: touchforms/backend/test_server.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import unittest
+import sys
+from xcp import InvalidRequestException
+import xformserver
+import xformplayer
+
+
+class DummyServer(xformserver.XFormHTTPGateway):
+    def __init__(self):
+        pass
```
Comment (czue) [inline touchforms/backend/test_server.py]: this is clever

File: touchforms/backend/xformserver.py [hunk 2]
```diff
@@ -70,22 +70,20 @@ def do_POST(self):
...
             if isinstance(e, java.lang.Exception):
                 e.printStackTrace()  # todo: log the java stacktrace
             elif isinstance(e, urllib2.HTTPError):
                 if e.headers.get("content-type", "") == "text/plain":
                     msg = e.read()
-            elif isinstance(e, TouchFormsException):
-                error_type = type(e).__name__
 
-            logging.exception('error handling request')
+            info = sys.exc_info()
```
Comment (esoergel) [inline touchforms/backend/xformserver.py]: What's the advantage of this over extracting information from `e`?

File: touchforms/backend/xformserver.py [hunk 3]
```diff
@@ -141,15 +139,25 @@ def do_GET(self):
...
+def ensure_required_params(params, action, content):
+    for param in params:
+        if param not in content:
+            raise InvalidRequestException("'%s' is required for action: %s" % (param, action))
+
+
 def handle_request(content, server):
     if 'action' not in content:
-        return {'error': 'action required'}
+        ensure_required_params(['action'], 'All actions', content)
```
Comment (esoergel) [inline touchforms/backend/xformserver.py]: woo! validating user input!
You can probably get rid of `if 'action' not in content:` here, right?

**Comment by esoergel** [comment]
Nice reworking!  Looks g2m when you're ready.

**Comment by snopoke** [comment]
nice, great to have some tests going here that we can build on!

**Comment by benrudolph** [comment]
Thanks for feedback will update. There's actually an error in this so please don't deploy. Touchcare api doesn't require session id

**Comment by benrudolph** [comment]
(Will fix once I get to the office)

File: touchforms/backend/xformserver.py (hunk 2 above)
Comment (benrudolph) [inline touchforms/backend/xformserver.py]: I was having trouble getting consistent information from the error message. if i raised a java.lang.Exception i'd have different methods than a regular python exception. How do you normally do it?

Comment (esoergel) [inline touchforms/backend/xformserver.py]: Sounds legit. I have no idea how one normally handles java exceptions in Jython.

**Pull request merged**